integration-tests/tests/utils/docker_utils.py

```diff
@@ -0,0 +1,61 @@
+"""Provide utility functions related to the use of Docker during integration tests."""
+
+import re
+import shutil
+import subprocess
+from enum import Enum
+
+
+class DockerStatus(str, Enum):
+    """Enum of possible Docker Statuses."""
+
+    created = "created"
+    restarting = "restarting"
+    running = "running"
+    removing = "removing"
+    paused = "paused"
+    exited = "exited"
+    dead = "dead"
+
+
+def get_docker_binary_path() -> str:
+    """
+    Finds the absolute path to the Docker client in the current PATH.
+
+    :return: Absolute path to the Docker binary.
+    :raise RuntimeError: docker is not found on PATH.
+    """
+    docker_bin = shutil.which("docker")
+    if docker_bin is None:
+        err_msg = "docker not found in PATH"
+        raise RuntimeError(err_msg)
+    return docker_bin
+
+
+def list_running_containers_with_prefix(prefix: str) -> list[str]:
+    """
+    Lists running Docker containers whose names begin with `prefix` and end with one or more digits.
+
+    :param prefix:
+    :return: List of running container names that match the pattern.
+    """
+    docker_bin = get_docker_binary_path()
+
+    # fmt: off
+    docker_ps_cmd = [
```

Comment:
@coderabbitai
Open an issue that we will switch to use `docker compose ps` with project name specified to check for the exact list of running containers. It can also check for containers that should've exited at certain stages of the package run.